src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. Since `Node` is an inner class, you don't need to pass `null` for `prev` and `next`, just pass the appropriate references.